assets/js/script.js

```diff
@@ -133,87 +163,174 @@ function timerStart() {
-    var q = questionLength;
-    if (q < 20) {
-        return 5;
-    } else if (q < 50) {
-        return 10;
-    } else if (q < 75) {
-        return 15;
-    } else if (q < 200) {
-        return 20;
+/* 
+ * Function that returns the amount of seconds of delay based on the length of the question in characters. 
+ * Used to hold the timer and allow the player to read the question.
+ */
+
+function calculateDelay(qLength) {
+    if (qLength < 21) {
+        return 2;
+    } else if (qLength < 51) {
+        return 4;
+    } else if (qLength < 100) {
+        return 6;
+    } else if (qLength < 200) {
+        return 8;
     }
 }
 
-// Wrong answer function
+/* 
+ * Function to display Winner pop up when all question answered correctly
+ */
+
+function victory() {
+    $('#victoryWindow').modal('show');
+}
+
+/*
+ * Wrong answer function
+ */
+
 function wrongAnswer() {
-    alert('Sorry, wrong answer\nCorrect answer is ' + selection.cor);
     $('#' + selection.cor).addClass('correct-answer');
     gameOver();
 }
 
-// Game over function
+/*
+ * Game over function, used when player clicks on wrong answer
+ */
 
 function gameOver() {
-    step = step - 1;
-    resetTimer();
-    saveScores();
-    $('#countdown').hide();
-    $('#step').text(step + 1);
-    $('#prizeWon').text(prize[step]);
-    $('#gameOverWindow').modal('show');
+    if (step === 0) {
+        resetTimer();
+        $('#countdown').hide();
+        $('#step').text(`You've failed to answer the first question`);
+        $('#prizeWon').text(`You didn't win any money :-(`);
+        $('#gameOverWindow').modal('show');
+    } else {
+        resetTimer();
+        saveScores();
+        $('#countdown').hide();
+        $('#step').text(`You've reached question ${step}`);
+        $('#prizeWon').text(`Your prize is ${prize[step-1]}`);
+        $('#gameOverWindow').modal('show');
+    }
 }
 
-function playAgain() {
+/* 
+ * Function called when game fails either by out of time event or by selecting the wrong question
+ */
+
+$('.playAgainButton').click(function() {
     resetTimer();
     setInterval(function() {
         location.reload();
-    }, 2000);
-}
+    }, 700);
+});
 
-// Time Out function
+/*
+ * Time Out function, used when player doesn't click on any answer
+ */
 
 function timeOut() {
-    $('#countdown').hide();
+    $('#countdown').html('<i class="fas fa-hourglass-half"></i> 0');
     $('#timeOutWindow').modal('show');
 }
 
-// Save scores function
+/* 
+ * Save scores function, only saving results with more than 3 questions answered correctly
+ */
 
 function saveScores() {
-    var score = {
-        step: step,
-        score: prize[step],
-        player: player,
-    };
-    highScores.push(score);
-    highScores.sort((a, b) => b.step - a.step);
-    highScores.splice(10);
+    if (step >= 3) {
+        let score = {
+            step: step,
+            score: prize[step],
+            player: player,
+        };
+        highScores.push(score);
```

Comment:
NIce to have feature:
Check local storage first, if player name is already present, then modify it otherwise enter new record. So we don't see same player name twice on the list